src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs

```diff
@@ -105,7 +105,7 @@ public class CommandLineCmdletBase : PSCmdlet, IDisposable
         /// </summary>
```

Comment:
@vexx32, your last commit had 1 failures in `PowerShell-CI-windows`
Remoting loopback tests.Can execute command in a disconnected session
```powershell
Objects cannot be added to a closed buffer. Make sure the buffer is open for Add and Insert operations to succeed.
at <ScriptBlock>, D:\a\1\s\test\tools\Modules\HelpersRemoting\HelpersRemoting.psm1: line 226
226:     Invoke-Command @parameters

```

Reply:
Okay, I've been through and tried to debug this locally and come up with nothing. I'm going to kick it again and see if it's just transient and I've been unlucky.

---

I have attempted to run the tests locally to reproduce it, but I can't reproduce it on either of my own Windows machines.

I _did_ find a completely different issue, though; the tests seem to always use an implicit credential, which I suspect may be causing problems with the loopback remoting restrictions in place in more recent Windows versions. The following error is given when I try to run the tests with their default configuration:
```ps1
VERBOSE: Using Implicit Credential
New-PSSession: C:\Users\joelp\source\repos\PowerShell\test\tools\Modules\HelpersRemoting\HelpersRemoting.psm1:191
Line |
 191 |          $session = New-PSSession @parameters
     |                     ~~~~~~~~~~~~~~~~~~~~~~~~~
     | [localhost] Connecting to remote server localhost failed with the following error message : WinRM
     | cannot process the request. The following error with errorcode 0x8009030e occurred while using
     | Negotiate authentication: A specified logon session does not exist. It may already have been
     | terminated.    Possible causes are:   -The user name or password specified are invalid.   -Kerberos is
     | used when no authentication method and no user name are specified.   -Kerberos accepts domain user
     | names, but not local user names.   -The Service Principal Name (SPN) for the remote computer name and
     | port does not exist.   -The client and remote computers are in different domains and there is no trust
     | between the two domains.  After checking for the above issues, try the following:   -Check the Event
     | Viewer for events related to authentication.   -Change the authentication method; add the destination
     | computer to the WinRM TrustedHosts configuration setting or use HTTPS transport.  Note that computers
     | in the TrustedHosts list might not be authenticated.    -For more information about WinRM
     | configuration, run the following command: winrm help config. For more information, see the
     | about_Remote_Troubleshooting Help topic.  Other Possible Cause:   -The domain or computer name was not
     | included with the specified credential, for example: DOMAIN\UserName or COMPUTER\UserName.
```

Given that this looks like it could be more related to machine configuration than anything else, I modified my TrustedHosts a bit to remove that as a possible cause (they failed in this same way again) and then inserted an explicit credential to use into the tests so they would run.

With that tweak made, the test passed flawlessly ten consecutive times.

I'm not sure what's going on here, but I'm pretty confident it's not due to my code. Given the way that `Invoke-Command` is being run in the tests, this error has to occur during its `BeginProcessing` or `ProcessRecord` routines, which I have not touched.

I've also seen this issue appear in other PRs as well, though less consistently than it's showing up here.

Reply:
Seems like this might be a bit of a race condition issue. After some refactoring, this issue disappeared. 🤷